### ISSUE TITLE
Help me carry: Change to Single Try  + Restart

### ISSUE THREAD

**Issue by kyordhel** [RoboCupAtHome/RuleBook#263]
After GermanOpen I realized that *Help me carry* is not working as expected. No team will be able to complete the test in five minutes (although a human could).

Increasing the time to 8~10 minutes breaks possible pipe-lining with other tests taking 5 minutes, and makes hard or impossible to schedule such a long test and run it three times for 10+ teams (at least 5 hours).

The solution that comes to my mind is change the test schema from *3 Tries* (like most test in Stage 1) to the traditional **Single Try  + Restart**. This change makes possible to run the test for 16 teams (theoretical maximum per league) in about 2.5 hours (or just two hours if the test time is increased to 8 minutes).

**Comment by LoyVanBeek** [comment]
At the German Open, team Homer started the guiding phase when time ran out, IIRC. I think they would have made the full task in 5 minutes. 
The rules already allow us to reduce the number of attempts BTW. 

But please make this change in a separate pull-request as it is a separate issue.

**Comment by balkce** [comment]
Adding a restart is, in my opinion, worthless, since the issue here is time limit, and the time after the restart does not reset. So if a team restarts in minute 2, they would only have 3 minutes to complete the test, undermining this whole argument.

It also depends on how complex the arena is. In the Mexican Nationals, of the two teams that attempted the test, one was able to complete it. The other didn't but for reasons other than the time limit. Something tells me that this was because our arena (inside and out) was far simpler than yours.

I frankly like the idea of repeating the tests, and I'm pretty sure the teams do as well.

If you're still worried, amp the time to 8 minutes and decrease the attempts to two for this test, averaging both scores for the final score. That will take less than 3 hours.

About breaking the pipe-lining, I've never liked that strategy: it unnecessarily puts pressure on the judges and the teams and, if not handled right, is a logistical nightmare (moving the robot between arenas, not able to help the other judges because we are evaluating different tests, etc.).

I say: do one round of attempts after another, and maybe a small 10-minute break between round of attempts (A then B then C then break then A then B then C then break, etc) for the same test in the same arena. Since they're around 10 teams per arena, that gives teams around an hour to make changes between attempts. You can tweak the arena at every break, so each round of attempts is not the same as the last. If one judge needs help, another can jump in quickly since he/she has been evaluating the same test in the other arena. And in this way, all teams from the same league are working in their separate arena, but doing their variation of the test.

We did this in the Mexican Nationals and it worked well. To be fair, there were only four teams, but it felt comfortable for both the judges and the teams.

**Comment by kyordhel** [comment]
@balkce The idea is from 3x5'=15' change it to 1x10'=10', so we reduce the test by 5 minutes. Teams have a restart which means two chances in the same time.

About pipelining, and almost mandatory with 12+ teams (possibly 16). There will be no arena change since is one arena per league. German Open had I think 8 and went smooth, but doubling this number...

**Comment by balkce** [comment]
It's still a single try: the robot screws up at the 6-minute mark and the "second chance" is only 4 minutes. And I thought this whole issue was born out of thinking that 5 minutes was too short a time. If this isn't the case anymore, why not do 2x5'=10'? It is basically the same thing as 1x10'=10' with the added bonus that teams have some time between attempts to correct stuff.

I didn't understand your point about pipelining: if there is one arena per league, how is the pipelining going to work? What I understand by "pipelining" is the way that we did it in Germany last year: one test in arena A and then hop to Arena B for the other test. This isn't the case anymore... something tells me that what I understand by "pipelining" and what you mean by it are two different things.

**Comment by LoyVanBeek** [comment]
The basic HelpMeCarry route for GermanOpen was https://github.com/RoboCupAtHome/GermanOpen2017/blob/master/referee_definitions/HelpMeCarry/route_and_obstacles.png
The distance to between start/finish was less than 30 meters. The robots had to do this 3 times of course so lets round that to 100 meters of driving in 5 minutes. That's 0.33 m/s *on average*. 

I  think 10 minutes is too long but 5 too short. Maybe 7? If we run that 2x by default, we save a bit of time.

**Comment by HideakiNagano21** [comment]
Same in Japan Open. 
Especially for DSPL, there were no team to complete the following phase.
The teams did not have much time to prepare the coding for DSPL because HSR was new robot for them. 
HSR have maximal limit speed of moving and then almost all the team could not keep following the operator. 

How about permitting the custom operator and the free walk for DSPL? 
Of course, in that case, the score must be down point (for example -50%).

**Comment by LoyVanBeek** [comment]
What do you mean with free walk? 

Also, I took a look at the[ recorded data](https://we.tl/ioBdOOjzDb) from Homer's most successful run: they took about 100s seconds to follow me along those 30 meters.

**Comment by HideakiNagano21** [comment]
Free walk means the unnaturally slow and backward walk for facing the robot.
I think it can be accepted with much down point. 
The audience would like to see the robot moving. Also TOYOTA and Softbank.

**Comment by kyordhel** [comment]
### Number of tries
The standard this far is the traditional *single try + restart* or *3 tries*, and I would like to leave it like this. If we have test that run once, twice, or trice, the rulebook will become way too chaotic, so I would prefer to leave the test as is.

### Pipelining
Since each league has its own arena, pipelining must be done in a single arena using half of it for one test, and half of it for the other. With a proper setup, I don't see why *Speech Recognition* and *Storing the Groceries* cant be run at the same time one in the living room and the other in the kitchen. *Help me carry* is tricky because the robot needs to move in and out, so it can't be run in parallel with GPSR, but it could with the others as long as they last the same amount of time.

**Comment by HideakiNagano21** [comment]
Allow me open this again and comment. 
I understand that the free walk is not admitted.
But please think about the limited speed of HSR, which maximum speed is 0.8 km/h only.
Thus, HSR can NOT follow the professional walker walking with natural speed for human. 
So, whoever the professional walker is, the walking speed should be less than 0.8 km/h.

**Comment by kyordhel** [comment]
This issue is about changing *Help me carry* to *Single Try + Restart*. Let's discuss *free walk* in another issue.
